src/main/java/com/yanna/stepanova/repository/BookRepository.java

```diff
@@ -2,9 +2,9 @@
 
 import com.yanna.stepanova.model.Book;
 import java.util.List;
+import java.util.Optional;
+import org.springframework.data.jpa.repository.JpaRepository;
 
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
+public interface BookRepository extends JpaRepository<Book, Long> {
+    Optional<List<Book>> findAllByAuthorContainsIgnoreCase(String author);
```

Comment:
No need to put list into Optional. We handle null with Optional. findAllByAuthorContainsIgnoreCase should never return null, if there are no books, findAllByAuthorContainsIgnoreCase returns empty list
```suggestion
    List<Book> findAllByAuthorContainsIgnoreCase(String author);
```